recipes/openssl/1.x.x/conanfile.py

```diff
@@ -447,58 +474,51 @@ def _tool(self, env_name, apple_name):
-                if "RANLIB" in os.environ:
-                    tools.replace_in_file(makefile_org, "RANLIB= ranlib\n", "RANLIB= %s\n" % self._adjust_path(os.environ["RANLIB"]))
-                rc = os.environ.get("WINDRES", os.environ.get("RC"))
-                if rc:
-                    tools.replace_in_file(makefile_org, "RC= windres\n", "RC= %s\n" % self._adjust_path(rc))
-                if "NM" in os.environ:
-                    tools.replace_in_file(makefile_org, "NM= nm\n", "NM= %s\n" % self._adjust_path(os.environ["NM"]))
-                if "AS" in os.environ:
-                    tools.replace_in_file(makefile_org, "AS=$(CC) -c\n", "AS=%s\n" % self._adjust_path(os.environ["AS"]))
-
-    def _get_env_build(self):
-        if not self._env_build:
-            self._env_build = AutoToolsBuildEnvironment(self)
-        return self._env_build
+        makefile_org = os.path.join(self.source_folder, "Makefile.org")
+        if not "CROSS_COMPILE" in os.environ:
+            cc = os.environ.get("CC", "cc")
+            gen_info = json.loads(load(self, os.path.join(self.generators_folder, "gen_info.conf")))
+            replace_in_file(self, makefile_org, "CC= cc\n", "CC= %s %s\n" % (self._adjust_path(cc), gen_info["CFLAGS"]))
+            if "AR" in os.environ:
+                replace_in_file(self, makefile_org, "AR=ar $(ARFLAGS) r\n", "AR=%s $(ARFLAGS) r\n" % self._adjust_path(os.environ["AR"]))
+            if "RANLIB" in os.environ:
+                replace_in_file(self, makefile_org, "RANLIB= ranlib\n", "RANLIB= %s\n" % self._adjust_path(os.environ["RANLIB"]))
+            rc = os.environ.get("WINDRES", os.environ.get("RC"))
+            if rc:
+                replace_in_file(self, makefile_org, "RC= windres\n", "RC= %s\n" % self._adjust_path(rc))
+            if "NM" in os.environ:
+                replace_in_file(self, makefile_org, "NM= nm\n", "NM= %s\n" % self._adjust_path(os.environ["NM"]))
+            if "AS" in os.environ:
+                replace_in_file(self, makefile_org, "AS=$(CC) -c\n", "AS=%s\n" % self._adjust_path(os.environ["AS"]))
 
     def _get_default_openssl_dir(self):
         if self.settings.os == "Linux" and self._full_version >= "1.1.0":
             return "/etc/ssl"
-        return os.path.join(self.package_folder, "res")
+        return "res"
 
     @property
     def _configure_args(self):
         openssldir = self.options.openssldir or self._get_default_openssl_dir()
-        prefix = tools.unix_path(self.package_folder) if self._win_bash else self.package_folder
-        openssldir = tools.unix_path(openssldir) if self._win_bash else openssldir
+        openssldir = unix_path(self, openssldir) if self.win_bash else openssldir
         args = [
           '"%s"' % (self._target if self._full_version >= "1.1.0" else self._ancestor_target),
           "shared" if self.options.shared else "no-shared",
-                "--prefix=\"%s\"" % prefix,
+                "--prefix=/",
```

Comment:
This jumped out at me, this was the package folder but now it's just root 🤔 Is the new Autotools install doing a better job?

Reply:
Hi @prince-chrismc,.
It is my understanding that in Conan 2.,0 self.package_folder is determined dynamically in the export() method and is not known when the generate() or build() methods are called. This is the pattern I have been using to permit build artifacts to be relocated at export-pkg time. Let me know if I am missing anything. Thanks.